src/utils/eis_utils/src/eis_utils.c

```diff
@@ -418,36 +425,109 @@ EISErr BuildSasCertConnectionString(
+ * @details Calls into the EIS Identity and get the identities response, saves to a file
  * @param[in] timeoutMS the timeoutMS in milliseconds for each call to EIS
- * @param[out] provisioningInfo the pointer to the struct which will be initialized with the information for creating a connection to IotHub using the EIS supported provisioning information
- * @returns on success a null terminated connection string, otherwise NULL
+ * returns EISErr value
  */
-EISUtilityResult RequestConnectionStringFromEISWithExpiry(
-    const time_t expirySecsSinceEpoch, uint32_t timeoutMS, ADUC_ConnectionInfo* provisioningInfo)
+EISErr EISIdentitiesPipeWriter(uint32_t timeoutMS, const char* pipePath)
 {
-    EISUtilityResult result = { EISErr_Failed, EISService_Utils };
+    char* identityResponseStr;
 
-    if (provisioningInfo == NULL)
+    EISErr retCode = EISErr_Ok;
+
+    EISErr identityResult = RequestIdentitiesFromEIS(timeoutMS, &identityResponseStr);
+
+    if (identityResponseStr != NULL)
     {
-        result.err = EISErr_InvalidArg;
-        return result;
+        int pipeFd = open(pipePath, O_WRONLY);
+        if (pipeFd < 0)
+        {
+            retCode = EISErr_NamedPipeFailure;
+            goto done;
+        }
+        // dprintf(pipeFd, "%s\n", identityResponseStr);
+        // dprintf(pipeFd, "%d\n", identityResult);
+        write(pipeFd, identityResponseStr, strlen(identityResponseStr));
+        write(pipeFd, "\n", 1);
+
+        char identityResultStr[32];
+        snprintf(identityResultStr, sizeof(identityResultStr), "%d\n", identityResult);
+        write(pipeFd, identityResultStr, strlen(identityResultStr));
+
+        close(pipeFd);
+
+        printf("Connection data written to named pipe %s.\n", pipePath);
     }
+    else
+    {
+        retCode = EISErr_ConnErr;
+    }
+done:
+    return retCode;
+}
 
-    if (expirySecsSinceEpoch <= time(NULL) || timeoutMS == 0)
+/**
+ * @brief Reads EIS Identities information from a EIS data file
+ * @details Reads EIS Identity from a file, first line being the identity response, 
+ * and the second line being the EIS Error Code.
+ * @param[in,out] identityResponseBuffer the pointer to Identity response string
+ * @param[in,out] childErrno the errno to bubble up to the caller
+ * @returns a value of EISErr
+ */
+EISErr EISIdentitiesPipeReader(const char* pipePath, char** identityResponseBuffer, int* childErrno)
+{
+    EISErr result=0;
+    int pipeFd = open(pipePath, O_RDONLY);
+    if (pipeFd < 0)
     {
-        result.err = EISErr_InvalidArg;
+        *childErrno = errno;
         return result;
     }
 
+    *identityResponseBuffer = malloc(1024 * sizeof(char));
+    if (*identityResponseBuffer == NULL) {
+        *childErrno = errno;
+        goto done;
+    }
+    
+    if (read(pipeFd, *identityResponseBuffer, 1024) < 0) {
+        *childErrno = errno;
+        free(*identityResponseBuffer);
+        goto done;
+    }
+
+    char readidentityResult[1024];
```

Comment:
please declare variables at beginning of the block, also for line 508. It's compiling because of gnu compiler extensions, which could cause compat issues with other compilers in the future (msvc, clang, etc).  We can fix it overall by adding -pedantic cflag.